Sources/protoc-gen-swift/FileGenerator.swift

```diff
@@ -96,7 +96,7 @@ class FileGenerator {
           }
         }
 
-        p.print("\(comments)import Foundation")
+        p.print("\(comments)\(generatorOptions.importDirective.snippet) Foundation")
```

Comment:
At first glance, does it make sense for this to end up as `public import` or `@_implementationOnly import`?  i.e. - I can see why it should get a qualifier, but should it be the same as all the proto modules, or should it be a fixed/specific value?

Reply:
We do expose properties of type `Data` when the generated visibility is public, so in those scenarios we do need this to be `public import`. This is why I went for using the import directive (so that we don't import it as public if we are generating with a different visibility).

However, a downside of this approach is that marking an import as `public` will generate warnings if it's not actually used in public API in the file - and this _can_ happen in generated code, because we're **always** importing Foundation. 

Ideally, we'd have some logic to check whether we actually need `Foundation` or not in a given file, and decide whether to include the import at all or not. But this sounds like a bit of a pain to do (or like it would be pretty slow if e.g. we'd check each generated file at the end to see if we've used `Data` or something similar)

Reply:
ack, tx.

I'd suggest opening an issue to have the generator conditional add that import (I guess we'd just need it for `bytes` fields on a message or extension)

Reply:
https://github.com/apple/swift-protobuf/issues/1702